### PR TITLE
git: add basic support for LFS

### LINKED ISSUE #542: Support git lfs
Git LFS (large file support) is usually installed by running `git lfs install` which tries to modify `~/.gitconfig`. This is incompatible with home manager if `programs.git.enable = true;`.

If I understand it correctly, LFS just needs to set the following config:

```
[filter "lfs"]
	smudge = git-lfs smudge -- %f
	process = git-lfs filter-process
	required = true
	clean = git-lfs clean -- %f
```

So there should maybe be an option `programs.git.lfs.enable = true;` that sets this config and installs `git-lfs`?

### PR DESCRIPTION
**Pull request by rycee**
Fixes #542

### CODE REVIEW

File: modules/programs/git.nix [hunk 1]
```diff
@@ -133,6 +133,10 @@ in
         '';
         description = "List of configuration files to include.";
       };
+
+      lfs = {
+        enable = mkEnableOption "Git Large File Storage";
```
Comment (worldofpeace) [inline modules/programs/git.nix]: We should probably add an option `skipSmudge` which does the following

>  Skips automatic downloading of objects on clone or pull. This requires a
    manual "git lfs pull" every time a new commit is checked out on your
    repository.

I think lots of people, myself included, use this global configuration.

It just adds a `--skip` to the following:
```
smudge = git-lfs smudge --skip -- %f
process = git-lfs filter-process --skip
```

Comment (worldofpeace) [reply modules/programs/git.nix]: btw, you'd get this config if you issued `git lfs install --skip-smudge`.

Comment (rycee) [reply modules/programs/git.nix]: A good idea, I've added a `skipSmudge` option.

**Comment by rycee** [comment]
Thanks for trying it out! Rebased to master in 1cdb8abf301f5c5ac99849d3754d1477b5d2b3e5.

**Pull request closed without merging**